app/models/king.rb

```diff
@@ -1,2 +1,21 @@
 class King < Piece
-end
+	def valid_move?
+		# Find king's current position on board.
+		king = King.find_by(x, y)
```

Comment:
This code is being called from within a king object. So in this context, `self` is already the king that you are looking for. So this line isn't needed. :smile: